### PR TITLE
DEVOPS-95 Script for sending copies of backup files to ARIS HPC.

### PR DESCRIPTION
**Pull request by Tas-sos**
This is my script to sync backup files with data center.

### CODE REVIEW

File: rSync_data_with_dataCenter/rSyncBackups.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+echo -e "| Backup of files to host \e[91m\e[1m$REMOTE_HOST\e[97m\e[21m begins at \e[34m`date`\e[97m \t|"
+echo -e "--------------------------------------------------------------------------------------\n"
+
+/usr/bin/rsync -avzh --progress -e "ssh -o StrictHostKeyChecking=no -i $SSH_KEY -va" $FROM_DIR $REMOTE_USER@$REMOTE_HOST:$REMOTE_PARENT_DIR 
+result=$?
+
+if [ $result -eq 0 ]
+then
+	echo -e "\n-----------------------------------------------------------------------------------------"
+	echo -e "| Backup of files to host \e[91m$REMOTE_HOST\e[97m \e[92m\e[1mcompleted\e[97m\e[21m at \e[34m`date`\e[97m \t|"
```
Comment (themiszamani) [inline rSync_data_with_dataCenter/rSyncBackups.sh]: encoding

File: rSync_data_with_dataCenter/rSyncBackups.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		echo -e "[Usage]: ./rSyncBackups.sh -u user -r datacenter.gr -k ~/.ssh/supper-Key -f backup.tar.gz -t /home/debian/backups "
+		exit 0
+		;;
+	u) REMOTE_USER=${OPTARG} ;;
+	r) REMOTE_HOST=${OPTARG} ;;
+	k) SSH_KEY=${OPTARG} ;;
+	f) FROM_DIR=${OPTARG} ;;
+	t) REMOTE_PARENT_DIR=${OPTARG} ;;
+	esac
+done
```
Comment (themiszamani) [inline rSync_data_with_dataCenter/rSyncBackups.sh:64]: i think we need some default values. Maybe the remote host.

**Review by themiszamani — CHANGES_REQUESTED** [review]

File: rSync_data_with_dataCenter/rSyncBackups.sh (hunk 2 above)
Comment (Tas-sos) [reply rSync_data_with_dataCenter/rSyncBackups.sh:64]: Thank you for review Τhemis,

The reason I don't have a default value is that I designed and implemented it to make it as dynamic as possible.
But if you wanted default values,  no problem.

File: rSync_data_with_dataCenter/rSyncBackups.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+set -e
+set -u
+set -f
+set -o pipefail
+
+while getopts h:u:r:k:f:t: option 
+do
+	case "${option}" in
+	h|--help)
+		echo -e "rSync Backaps."
```
Comment (themiszamani) [inline rSync_data_with_dataCenter/rSyncBackups.sh]: small typo ;) Backups

**Review by themiszamani — COMMENTED** [review]
i dont see the multiple directories.

File: rSync_data_with_dataCenter/rSyncBackups.sh [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	f) FROM_DIR=(${OPTARG})	;;
+	t) REMOTE_PARENT_DIR=${OPTARG} ;;
+	esac
+done
+
+echo -e "\n---------------------------------------------------------------------------------"
+echo -e "| Backup of files to host $red$bold_on$REMOTE_HOST$bold_off$color_off begins at $blue`date`$color_off \t|"
+echo -e "---------------------------------------------------------------------------------\n"
+
+/usr/bin/rsync -avzh --progress -e "ssh -o StrictHostKeyChecking=no -i $SSH_KEY -va" ${FROM_DIR[@]} $REMOTE_USER@$REMOTE_HOST:$REMOTE_PARENT_DIR
```
Comment (Tas-sos) [inline rSync_data_with_dataCenter/rSyncBackups.sh]: @themiszamani 
Νow sends either one or all of the paths it has taken from the user.
Now the variable `${FROM_DIR[@]}` is a **array**.

**Pull request merged**